src/app/search/directives/story-list.component.ts

```diff
@@ -11,14 +11,24 @@ import { StoryModel } from '../../shared';
     <div *ngIf="!noStories">
       <div class="story-list">
         <publish-story-card *ngFor="let s of stories" [story]="s"></publish-story-card>
+        <div *ngFor="let i of emptyCards" class="empty-story-card"></div>
         <div *ngFor="let l of loaders" class="story"><publish-spinner></publish-spinner></div>
       </div>
     </div>
 `
 })
 
-export class StoryListComponent {
+export class StoryListComponent implements OnChanges {
   @Input() noStories: boolean;
   @Input() stories: StoryModel[];
   @Input() loaders: boolean[];
+
+  emptyCards: number[];
+
+  ngOnChanges(changes: any) {
+    if (changes['stories'] && changes['stories'].currentValue &&
+        changes['stories'].currentValue.length > 0 && changes['stories'].currentValue.length % 3 > 0) {
+      this.emptyCards = Array.from(Array(3 - changes['stories'].currentValue.length % 3), (x, i ) => i);
```

Comment:
The extra empty card gets wrapped but the zero height doesn't affect layout. Really glad you thought of this zero height empty cards idea because I did not want to handle component width/height. I don't know about viewport, but I tried checking component width or height for something else and it always came back zero.